### ISSUE TITLE
Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import

### ISSUE THREAD

**Issue by randomnessdev** [nuxt-community/firebase-module#602]
### Version
@nuxtjs/firebase:  7.5.0 -->
firebase: ex: 8.10.0
nuxt: 2.15.8

macOS Big Sur

### Reproduction Link
Cannot share publicly but can give access in private

### Steps to reproduce
``` 
npm run generate
``` 

### What is Expected?
Having an "npm run generate" working

### What is actually happening?

``` 
x Nuxt Fatal Error                                                          
Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import'/Users/phil/Documents/GitHub/project/node_modules/firebase/app' 
is not supported resolving ES modules imported from /Users/phil/Documents/GitHub/project/node_modules/@nuxtjs/firebase/lib/module.js
Did you mean to import 
/Users/phil/Documents/GitHub/project/node_modules/firebase/app/dist/index.cjs.js?  
```

I found [this solution](https://fredriccliver.medium.com/directory-import-node-modules-firebase-app-is-not-supported-resolving-es-modules-imported-843d9aeceb79) but i have no idea how to apply it to a Nuxt project

I tried many different versions of @nuxtjs/firebase & firebase, but issue always remain,

Can anyone help?

**Comment by randomnessdev** [comment]
Found solution:
First, added this to nuxt.config.js :

`hooks: {
  generate: {
    async done(builder) {
      const appModule = await import('./.nuxt/firebase/app.js')
      const { session } = await appModule.default(
        builder.options.firebase.config,
        {
          res: null,
        }
      )
      try {
        session.database().goOffline()
      } catch (e) { }
      try {
        session.firestore().terminate()
      } catch (e) { }
    },
  },
},
`

then switched "terminateDatabasesAfterGenerate: true" to false, 

then, to deploy it to Netlify, i removed the .nuxt folder from the gitignore

**Comment by mandalornl** [comment]
@randomnessdev @lupas changing the import to `firebase/compat/app`, instead of the absolute path seems to work. You can use the [patch](https://github.com/ds300/patch-package) below, if you'd like.

```javascript
diff --git a/node_modules/@nuxtjs/firebase/lib/module.js b/node_modules/@nuxtjs/firebase/lib/module.js
index d3aa31f..5e76a2b 100644
--- a/node_modules/@nuxtjs/firebase/lib/module.js
+++ b/node_modules/@nuxtjs/firebase/lib/module.js
@@ -326,9 +326,8 @@ function validateConfigKeys(options, currentEnv) {
  * See https://github.com/nuxt-community/firebase-module/issues/93
  */
 function terminateDatabasesInGenerateHooks(ctx) {
-  const firebaseDir = ctx.nuxt.options.srcDir + '/node_modules/firebase/'
   ctx.nuxt.hook('generate:before', async (generator) => {
-    const { default: firebase } = await import(firebaseDir + 'app')
+    const { default: firebase } = await import('firebase/compat/app')
 
     if (!firebase.apps.length) {
       firebase.initializeApp(generator.options.firebase.config)
```

**Comment by mandalornl** [comment]
@randomnessdev @lupas I've updated the patch in my comment above. I had to also import `firebase/compat/database` and `firebase/compat/firestore` for it to actually work.

**Comment by lupas** [comment]
Thanks guys, fixed it with [v8.1.0](https://github.com/nuxt-community/firebase-module/releases/tag/v8.1.0).

Unfortunately, the **terminateDatabasesInGenerateHooks()** function doesn't seem to do its job anymore since v8. The warning "_The command 'nuxt generate' finished but did not exit after 5s"_ this function was supposed to fix appears again. :/

<img width="653" alt="Screenshot 2021-11-30 at 07 28 09" src="https://user-images.githubusercontent.com/4020037/143998842-33c06ce2-9980-4437-8828-fb54763b5a37.png">

But that's not a biggie, for Nuxt v3 I'll have to re-write the module anyway.. :)

Thanks a lot!

**Comment by Strift** [comment]
Hello,

I know this is closed, but I think I've found a workaround for v7.x.

- Set `terminateDatabasesAfterGenerate` to `false`
- Implement the [fix to terminate database](https://firebase.nuxtjs.org/community/faq/#nuxt-generate-warns-with-nuxt-generate-finished-but-did-not-exit) in `generate:done` hook
- Update your `nuxt.config.js` file as following

```javascript
// nuxt.config.js

build: {
    extend (config) {
      config.resolve.alias['firebase/app'] = 'firebase/app/dist/index.cjs.js'
    }
}
```

I hope this can help.
